sdk/python/kfp/dsl/for_loop.py

```diff
@@ -188,6 +191,16 @@ def from_pipeline_channel(
         compilation progress in cases of unknown or missing type
         information.
         """
+        # TODO: if is LoopArgumentVariable, check if nested items are lists
+        if not isinstance(channel, LoopArgumentVariable):
+            type_name = type_annotations.get_short_type_name(
+                channel.channel_type)
+            parameter_type = type_utils.PARAMETER_TYPES_MAPPING[
+                type_name.lower()]
+            if parameter_type != type_utils.LIST:
+                raise ValueError(
+                    'Cannot iterate over a single Parameter using `dsl.ParallelFor`. Expected a list of Parameters as argument to `items`.'
```

Comment:
```suggestion
                    'Cannot iterate over a single parameter using `dsl.ParallelFor`. Expected a list of parameters as argument to `items`.'
```

Our error messages do not capitalize the "P" in parameter elsewhere in the SDK.